src/webpimage.cpp

```diff
@@ -508,10 +509,11 @@ namespace Exiv2 {
 #endif
 
         chunkId.pData_[4] = '\0' ;
-        while ( !io_->eof() && (uint64_t) io_->tell() < filesize) {
+        while (!io_->eof() && io_->tell() < filesize) {
             io_->readOrThrow(chunkId.pData_, WEBP_TAG_SIZE, Exiv2::kerCorruptedMetadata);
             io_->readOrThrow(size_buff, WEBP_TAG_SIZE, Exiv2::kerCorruptedMetadata);
-            const uint32_t size = Exiv2::getULong(size_buff, littleEndian);
+            const long size = Exiv2::getULong(size_buff, littleEndian);
+            enforce(0 <= size, Exiv2::kerCorruptedMetadata);
```

Comment:
This check relies on undefined behavior (signed integer overflow) and is thus not effective, as the compiler is free to optimize it away.